packages/next/src/server/web/spec-extension/adapters/next-request.ts

```diff
@@ -8,8 +8,7 @@ import { NextRequest } from '../request'
 
 export class NextRequestAdapter {
   public static fromBaseNextRequest(request: BaseNextRequest): NextRequest {
-    // TODO: look at refining this check
-    if ('request' in request && (request as WebNextRequest).request) {
+    if (request.constructor.name === 'WebNextRequest') {
```

Comment:
This could be a problem with minification.

Reply:
As discussed, we may have to rely on this for the time being due to bundling issues and the feasability of `instanceof`.